supplier/tests/test_supplier_page.py

```diff
@@ -1,8 +1,31 @@
 import pytest
-from unittest.mock import Mock
+from unittest.mock import Mock, call
 from django.contrib.auth.models import User
 from supplier.models import Supplier
-from supplier.views import get_supplier
+from supplier.views import action_display_all_products, get_supplier, receive_product, action_search_product
+from django.test import RequestFactory
+from supplier import views
+from supplier_product.models import SupplierProduct
+from product.models import Product
+
+
+@pytest.fixture
+def searched_product_string():
```

Comment:
fixtures should be in conftest

Reply:
i dont mind putting all the fixtures in the conftest, thou i wouldnt think thats a good idea, the conftest file should have fixtures that are used all over the project, fixture like the one you mentioned or any other fixtures that was created for specific test, in my idea, shouldnt be in the general conftest... dont mind bringing it up for the all team to decide.

Reply:
I'm with your opinion Paz

Reply:
If you have a fixture for specific test like "abc",why not just using the "abc" instead of creating a fixture,I agree on what is more comfortable for you its just an oppinion.

Reply:
its the principle of not using "magic words".
usually you will make those kinds of variable as const defines (like in cpp) but since python doesnt have const, fixtures are the closest to it.
if you use global variable it might be changed in one of the test that it was used, using fixture the variable always created once it is called.

Reply:
You can have a file of constants and each constant will be informat like:
 PI = 3.14
so every one will be aware of it even tho you have created a variable in your test no one can override it and even if they created the same name in other file but its okay it was just an opinion

Reply:
`PI = 3.14`
that considered to be global variable, its not a const.

Reply:
File of constants that is used when you want a variable from there,since our goal is just to access the vars then that makes them constant for us,i know that python doesnt support constants in a direct way but if you create a folder in the root for an example called constants and it has every global variable in the files then it is easier and more understandable instead of a feature that is returning 5 or "a" you can save that to a variable in that file with better name.

Reply:
but if some one changes the variable during a test? even by mistake?

Reply:
What variable and where,this can happend everywhere i can change things in conftest by mistake during a test and cause tests to fail,i dont understand really what is the question

Reply:
I don't think adding a file with the constants is a good idea. It increases the complexity of the test procedure and reduces its visibility.

My rule of thumb for simple fixtures like that: if you need a value to be identical in more than one place (e.g. a fixture and a test that relies on that fixture), then create a fixture. If you just need it inside a test (it could be multiple tests as well), then you can either use the value directly or create a local variable.

I do think the fixtures should be declared according to the scope they are needed. If it's something that is very specific to the test module, then no need to introduce that into a wider scope -- i.e. higher-level conftest file.

In any case, I wouldn't spend too much time on that at the moment.